sycl/test/atomic_ref/accessor.cpp

```diff
@@ -0,0 +1,59 @@
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <numeric>
+#include <vector>
+using namespace sycl;
+using namespace sycl::intel;
+
+// Equivalent to add_test from add.cpp
+// Uses atomic_accessor instead of atomic_ref
+template <typename T>
+void accessor_test(queue q, size_t N) {
```

Comment:
I would be a tiny bit more confident, if `atomic_accessor` be at least instantiated with `local` target.

Since accessor with local target is a dedicated specialization - something may go wrong in a very subtle way.

Reply:
Added a test in 3e86ca2011d7fdbc2c57d14fd137384bbcda42da and some minor changes to the local `accessor` implementation in dcb259e4c811683c38dfedd3f65539276e957dff.

The fact that it didn't work out of the box surprised me, so thanks for suggesting that I add a test.  Moving some of the local `accessor` members from `private` to `protected` feels like a little bit of a hack, but it also seems reasonable to me that all standard accessors should expose the same interfaces if we're expecting developers to extend them.

The fact that I couldn't find an obvious way to extend the `accessor` class without using some of these hidden internal functions in the first place might suggest we need to revisit the public interface.  I think things are better in SYCL 2020 provisional (e.g. there's a decorated `get_multi_ptr` to replace `getQualifiedPtr`) but the unspecified behavior of `operator[]` is unfortunate.

Tagging @gmlueck, because I know he's been thinking a lot about accessors recently.